datasets/ir/business_registry/crawler.py

```diff
@@ -0,0 +1,66 @@
+from lxml import etree
+from lxml.html import HtmlElement
+from normality import slugify
+from typing import Generator, Dict
+from urllib.parse import urljoin
+
+from zavod import Context, helpers as h
+from zavod.shed.zyte_api import fetch_html
+
+
+def unblock_validator(doc: etree._Element) -> bool:
+    return doc.find(".//div[@class='o-grid']") is not None
+
+
+def crawl(context: Context):
+    page_number = 0
+    pages_processed = 0
+
+    while True:
+        # Construct the URL for the current page
+        url = f"https://www.unitedagainstnucleariran.com/iran-business-registry?page={page_number}"
+        context.log.info(f"Fetching URL: {url}")
+
+        # Fetch the HTML and get the table
+        doc = fetch_html(context, url, unblock_validator, cache_days=3)
+        table = doc.find(".//div[@class='view-content']//table")
+        if table is None:
+            context.log.info("No more tables found.")
+            break
+
+        # Iterate through the parsed table
+        for row in h.parse_html_table(table, skiprows=1):
+            str_row = h.cells_to_str(row)
+
+            company_elem = row.pop("company_sort_descending")
+            # Capture the link from the anchor tag
+            company_link = urljoin(
+                url, company_elem.find(".//a").get("href", "").strip()
+            )
+            withdrawn_elem = row.pop("withdrawn")
+            is_withdrawn = withdrawn_elem is not None
+
+            company_name = str_row.pop("company_sort_descending")
+            nationality = str_row.pop("nationality")
+            stock_symbol = str_row.pop("stock_symbol")
+
+            # Create and emit an entity
+            entity = context.make("Company")
+            entity.id = context.make_id(company_name, nationality)
+            entity.add("name", company_name)
+            entity.add("country", nationality)
+            entity.add("sourceUrl", company_link)
```

Comment:
do we wanna crawl the page too? Looks like we can get a company website there. not sure how useful that is.

would need to catch and ignore errors like this page https://www.unitedagainstnucleariran.com/company/540-sports-and-entertainment

Reply:
I think crawling the subpages for a description (notes) and website url could be good!